Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.h>
+
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/Serialization/Utils.h>
+#include <AzCore/IO/GenericStreams.h>
+
+#include <QMimeData>
+#include <QUrl>
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/SourceAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/ProductAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/FolderAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/RootAssetBrowserEntry.h>
+
+namespace AzToolsFramework
+{
+    namespace AssetBrowser
+    {
+        namespace Utils
+        {
+            void ToMimeData(QMimeData* mimeData, const AZStd::vector<const AssetBrowserEntry*>& entries)
+            {
+                if (!mimeData)
+                {
+                    return;
+                }
+                if (entries.empty())
+                {
+                    return;
+                }
+
+                AZStd::string outputString = ToString(entries);
+
+                // capture the URLs (full paths to items) in case you're about to drop this mimedata
+                // onto something which does not understand "asset browser entry" but does understand URLs.
+                QList<QUrl> urls;
+
+                for (const AssetBrowserEntry* entry : entries)
+                {
+                    if (entry)
+                    {
+                        urls.push_back(QUrl::fromLocalFile(QString::fromUtf8(entry->GetFullPath().c_str())));
+                    }
+                }
+                // Saving the buffer of asset browser entries back to the mime data
+                QByteArray byteArray(outputString.c_str(), static_cast<int>(outputString.size()));
+                mimeData->setData(AssetBrowserEntry::GetMimeType(), byteArray);
+                mimeData->setData("text/plain", byteArray);
```

Comment:
We set text/plain here so that you can drag an asset into Discord or an email or some other file and get its UUID, subid, name, etc.